Automattic/WP-Job-Manager
# Update date methods and expire jobs at the end of the day

Fixes #919

### Changes proposed in this Pull Request

* Update the date methods used and standardize all date rules to use the WP timezone.
* Expire jobs at the end of the day. Introduced filter `job_manager_jobs_expire_end_of_day` to revert to the original (expire at the start of the expiration day). We can easily flip this back.

We might have some changes to make in Application Deadline, but I don't _think_ anything urgent.

### Testing instructions

* Make sure dates are saved correctly when approving jobs and editing from backend. (Note: Don't forget you can't clear a job expiration if duration is set in settings)
* Dates appear correctly in metabox and on Job Dashboard page.
* In the database directly, have a job expire (`_job_expires` post meta) yesterday or today with `add_filter( 'job_manager_jobs_expire_end_of_day', '__return_false' );`. Run the `job_manager_check_for_expired_jobs` job manually and make sure they are set to a post_status=expired.


### New/Updated Hooks
- `job_manager_jobs_expire_end_of_day`
